package.json

```diff
@@ -7,6 +7,7 @@
   },
```

Comment:
> It seems like there are a lot of places where we're using ! without any reason to be sure that values are actually defined and non-null; I'm not clear on why this is an improvement from a type-safety standpoint?
>
> *-- @duxovni, https://github.com/matrix-org/matrix-js-sdk/pull/2835#pullrequestreview-1163892857*

What's the thoughts/summary here?

Reply:
The upshot from the individual comments was basically, we've got a lot of messy stuff with overloads where it'd be a pain to convince typescript that we're doing the right thing, so it's best to just tell typescript it's fine so we can go ahead and enable strict mode globally which will make life easier in general.

Reply:
And I assume we should avoid this situation more in future code if possible?

Do our current overloads doom us to `!` usage forever? Is there certain overload patterns we should avoid in the future?

Reply:
Overloads with different combinations of optionals, e.g. `A is optional & B is mandatory` | `A is mandatory & B is optional` will either need an if-case per combination to handle or `!`s